workflow/deterministic_wrappers.go

```diff
@@ -102,11 +102,11 @@ func Await(ctx Context, condition func() bool) error {
 // The following code will block until the captured count
 // variable is set to 5, or one hour passes.
 //
-//	workflow.AwaitWithTimeout(ctx, time.Hour, func() bool {
+//	workflow.AwaitWithTimeout(ctx, time.Hour, options, func() bool {
 //	  return count == 5
 //	})
-func AwaitWithTimeout(ctx Context, timeout time.Duration, condition func() bool) (ok bool, err error) {
-	return internal.AwaitWithTimeout(ctx, timeout, condition)
+func AwaitWithTimeout(ctx Context, timeout time.Duration, options TimerOptions, condition func() bool) (ok bool, err error) {
```

Comment:
Does changing this function signature break anything/anyone external?

Reply:
Yes it would break all callers when they update their SDK. That is not okay. Go lang doesn't give us a lot of tools to work with here so we need to just add a new API like `AwaitWithOptions` or `AwaitWithTimeoutAndOptions`

Reply:
had a hunch that'd be the case